Corgibytes.Freshli.Cli/Services/AgentReader.cs

```diff
@@ -9,31 +11,35 @@ namespace Corgibytes.Freshli.Cli.Services;
 // Once we have an actual agent we can use, then we can start processing the data from it.
 public class AgentReader : IAgentReader
 {
-    public List<Package> ListValidPackageUrls(string agentExecutable, PackageURL packageUrl) => agentExecutable switch
+    public List<Package> ListValidPackageUrls(string agentExecutable, PackageURL packageUrl)
     {
-        "/usr/local/bin/freshli-agent-csharp" => ReturnListOfPackages(packageUrl),
-        _ when agentExecutable.Contains("freshli-agent-test") => ReturnListOfPackages(packageUrl),
-        "/usr/local/bin/freshli-agent-javascript" => new(),
-        "/usr/local/agents/bin/freshli-agent-csharp" => new(),
-        "/home/freshli-user/bin/agents/freshli-agent-ruby" => new(),
-        _ => new()
-    };
-
-    private static List<Package> ReturnListOfPackages(PackageURL packageUrl) =>
-        packageUrl.Name switch
+        var package = new List<Package>();
+        var executablePath = Directory.GetCurrentDirectory() + Path.DirectorySeparatorChar + "bin" +
+                             Path.DirectorySeparatorChar + "freshli-agent-mock.rb";
+
+
+        var processInfo = new ProcessStartInfo
         {
-            "flyswatter" => new()
-            {
-                new(new("pkg:nuget/org.corgibytes.flyswatter/flyswatter@1.1.0"),
-                    new(1990, 11, 29, 0, 0, 0, TimeSpan.Zero))
-            },
-            "calculatron" => new()
-            {
-                new(new("pkg:nuget/org.corgibytes.calculatron/calculatron@21.3"),
-                    new(2022, 10, 16, 0, 0, 0, TimeSpan.Zero)),
-                new(new("pkg:nuget/org.corgibytes.calculatron/calculatron@14.6"),
-                    new(2019, 12, 31, 0, 0, 0, TimeSpan.Zero))
-            },
-            _ => new()
+            UseShellExecute = false,
+            FileName = executablePath,
+            Arguments = "validating-package-urls",
+            RedirectStandardOutput = true
         };
+        var process = Process.Start(processInfo); // Start that process.
+
+
+        while (process != null && !process.StandardOutput.EndOfStream)
+        {
+            var fileData = process.StandardOutput.ReadLine()?.Split(" ");
+            if (fileData != null)
+            {
+                package.Add(new(new(fileData[0]), DateTimeOffset.Parse(fileData[1])));
+            }
+        }
+
+        process?.WaitForExit();
```

Comment:
Jason added an abstraction for external CLI calls in https://github.com/corgibytes/freshli-cli/pull/140
We can use this to call the agent executable (I think you can pass "." as working directory). 
Then the output you get from that command can be processed like so:

```csharp
var packageUrlsWithDate = // Output from CLI Wrapped call
var packages = new List<Package>();

foreach (var packageUrlAndDate in packageUrlsWithDate)
{
    var separated = packageUrlAndDate.Split(" ");
    packages.Add(new(
        separated[0],
        DateTimeOffset.ParseExact(separated[1], "yyyy'-'MM'-'dd'T'HH':'mm':'ssK", null)
    ));
}
```

Reply:
why do we have to use the CliWrap if the process does the same thing? :)

Reply:
That's because of this PR: https://github.com/corgibytes/freshli-cli/pull/119
And this issue: https://github.com/corgibytes/freshli-cli/issues/51

For consistency I would prefer to stick to one method over introducing another way of talking to the CLI.